docs/1_python.md

```diff
@@ -0,0 +1,197 @@
+steps, and please open an issue if functionality is missing.
+
+## Installation 
+
+### Pypi
+
+The module is available in pypi as [oras](https://pypi.org/project/oras/)
+and you can install as follows:
+
+```bash
+$ pip install oras
+``` 
+   
+You can also clone the repository and install locally:
+
+```bash
+$ git clone https://github.com/oras-project/oras-py
+$ cd oras-py
+$ pip install .
+``` 
+
+Or in development mode:
+
+```bash
+$ pip install -e .
+```
+
+Development mode means that the install is done from where you've cloned the library,
+so any changes you make are immediately "live" for testing.
+
+### Docker Container
+
+We provide a [Dockerfile](https://github.com/oras-project/oras-py/blob/main/Dockerfile) to build a container with the client.
+
+```bash
+$ docker build -t oras-py .
+
+$ docker run -it oras-py                                                                                                                   
+# which oras-py
+/opt/conda/bin/oras-py
+```
+
+## Registry
+
+You should see [supported registries](https://oras.land/implementors/#docker-distribution) or if you
+want to deploy a local testing registry (without auth), you can do:
+
+
+```bash
+$ docker run -it --rm -p 5000:5000 ghcr.io/oras-project/registry:latest
+```
+
+To test token authentication, you can either [set up your own auth server](https://github.com/adigunhammedolalekan/registry-auth)
+or just use an actual registry. The most we can do here is set up an example that uses basic auth.
+
+
+```bash
+# This is an htpassword file, "b" means bcrypt
+htpasswd -cB -b auth.htpasswd myuser mypass
+```
+
+> :warning: The server below will work to login with basic auth, but you won't be able to issue tokens.
+
+```bash
+# And start the registry with authentication
+docker run -it --rm -p 5000:5000 \
+    -v $(pwd)/auth.htpasswd:/etc/docker/registry/auth.htpasswd \
+    -e REGISTRY_AUTH="{htpasswd: {realm: localhost, path: /etc/docker/registry/auth.htpasswd}}" \
+    ghcr.io/oras-project/registry:latest
+```
+
+## Command Line
+
+### Login
+
+Once you create (or already have) a registry, you will want to login. You can do:
+
+```bash
+$ oras-py login -u myuser -p mypass localhost:5000
+
+# or localhost (insecure)
+$ oras-py login -u myuser -p mypass -k localhost:5000 --insecure
+
+WARNING! Using --password via the CLI is insecure. Use --password-stdin.
+Login Succeeded
+```
+
+You can also provide them interactively
+
+        
+```bash   
+$ oras-py login -k localhost:5000 --insecure
+Username: myuser
+Password: mypass
+Login Succeeded
+```
+
+or use `--password-stdin`
+
+```bash
+$ echo mypass | oras-py login -u myuser -k localhost:5000 --insecure --password-stdin
+Login Succeeded
+```
+
+Note that oras-py will not remove content from your docker config files, so
+there is no concept of a "logout" unless you are using the client interactively,
+and have configs loaded, then you can do:
+
+```bash
+$ cli.logout(hostname)
```

Comment:
It does not look like a bash command. Should it be a block of python script?

Reply:
Yes good catch!